allennlp/data/tokenizers/sentence_splitter.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+from overrides import overrides
+from allennlp.common import Registrable
+from allennlp.common.util import get_spacy_model
+from allennlp.data.tokenizers.token import Token
+
+
+class SentenceSplitter(Registrable):
+    """
+    A ``SentenceSplitter`` splits strings into sentences. There are two variants of sentence splitting,
+    one that uses rules (ie punctuation), and another that uses a dependency parse of the text to infer
+    sentence boundaries.
+    """
+    default_implementation = 'spacy_rulebased'
+
+    def batch_split_sentences(self, texts: List[str]) -> List[List[Token]]:
+        """
+        This method lets you take advantage of spacy's batch processing.  
+        Default implementation is to just iterate of the sentences and call ``split_sentences``.
+        """
+        return [self.split_sentences(text) for text in texts]
+
+    def split_sentences(self, text: str) -> List[Token]:
+        """
+        Splits ``texts`` into a list of :class:`Token` objects.
+        """
+        raise NotImplementedError
+
+@SentenceSplitter.register('spacy_statistical')
+class SpacyStatisticalSentenceSplitter(SentenceSplitter):
+    """
+    A ``SentenceSplitter`` that uses spaCy's statistical sentence boundary detection.
+    More accurate but slower, since it uses a dependency parse to detect sentence boundaries.
+    """
+    def __init__(self,
+                 language: str = 'en_core_web_sm',
+                 pos_tags: bool = False,
+                 parse: bool = False,
```

Comment:
If this uses a dependency parse, how does it work if `parse=False`?

Reply:
It does fail if parse is False, I changed this so it sets `parse=True` if the user sets `rule_based=False`